### PR TITLE
modify supply/retrieve according to Change y to allowed_y logic

### PR DESCRIPTION
**Pull request by akayi07**

### CODE REVIEW

File: src/circom/newAmount1Calc_helper.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Field } from "delphinus-curves/src/field";
+import BN from "bn.js";
+
+export class NewAmount1CalcHelper {
```
Comment (yymone) [inline src/circom/newAmount1Calc_helper.ts:4]: Name can change to Amount1ToPoolCalcHelper if you still want to use class. (Or just do not use class, I suggest do not use class, as class should be concept for an aggregate concept) "New" is not meaningful from code logic.

File name should not use camel and snake case at the same time.

Function name can be calcAmount1ToPool, etc.
Parameter name is not properly. The amount1: BN can be changed to maxAllowedAmount1. The old_amount0 and old_amount1 should be poolAmount0 and poolAmount1, or liq0, liq1.

And, do not use snake and Camel case at the same time. For TS, we should use Camel case.

File: src/circom/ops/retrieve.ts [hunk 2]
```diff
@@ -36,7 +37,9 @@ export class RetrieveCommand extends Command {
     // STEP2: udpate liquility
     // circuits: check token0 != 0 || token1 != 0
     // circuits: liq0 >= amount0
-    // circuits: liq1 >= amount1
+    // circuits: liq1 >= amount1_new
+    const newAmount1Calc = new NewAmount1CalcHelper;
+    const amount1_new = newAmount1Calc.calcNewAmount1(amount0.v, amount1.v, liq0.v, liq1.v, false);
```
Comment (yymone) [inline src/circom/ops/retrieve.ts:42]: Please use meaningful parameter name.
Basically should change all old amount1 to allowed_min_amount1/allowed_max_amount1 for retrieve/supply.
And the calculated result can be amount1ToPool

Comment (akayi07) [reply src/circom/ops/retrieve.ts]: I changed the name.

File: src/circom/newAmount1Calc_helper.ts (hunk 1 above)
Comment (akayi07) [reply src/circom/newAmount1Calc_helper.ts]: I changed the name.

File: src/circom/amount1ToPoolCalcHelper.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Field } from "delphinus-curves/src/field";
+import BN from "bn.js";
+
+export function calcAmount1ToPool(
+    amount0: BN,
+    allowedAmount1: BN,
+    liq0: BN,
+    liq1: BN,
+    is_supply: boolean
```
Comment (yymone) [inline src/circom/amount1ToPoolCalcHelper.ts:9]: no snake case

Comment (akayi07) [reply src/circom/amount1ToPoolCalcHelper.ts]: I changed the name.

**Review by yymone — APPROVED** [review]

**Pull request merged**